### PR TITLE
Close #21974: No reason specified when attempting to place path additions on junctions

### LINKED ISSUE #21974: No reason specified when attempting to place benches, lamps, or bins on path with all four sides occupied
### Operating System

Windows 11 Home

### OpenRCT2 build

OpenRCT2, v0.4.11 (18d2b5e) provided by GitHub

### Base game

RollerCoaster Tycoon 2

### Area(s) with this issue?

_No response_

### Describe the issue

Fred's Workbench 2024-05-05 23-39-19
Fred's Workbench 2024-05-05 23-39-21
When attempting to place down new benches, lamps, or bins on footpath tiles which have all four sides connecting to other path (e.g. a path crossing or a tile in the middle of a 3x3 path grid), a "Can't position this here..." error message appears, but it does not provide a specific reason. A possible new error message reason in this case could be something like "No available sides for path additions on this footpath tile".
A similar issue existed with door wall scenery objects not providing a reason for being unable to be built over unsuitable track, but that was recently fixed in v0.4.11.

### Steps to reproduce

1. Build a path network including either at least a 3x3 path grid (in screenshot 1, shown using red tiled path) or a path crossing (in screenshot 2, shown using green tiled path)
2. Try to place down any benches, bins, or lamps on path tiles that have all four sides connected to other path tiles
3. No specific reason for the "Can't position this here..." message appears

### Attachments

_No response_

### PR DESCRIPTION
**Pull request by harvito**
when trying to add an edge path addition (bench, lamp, trashcan) to a path tile whose 4 edges are connected, the game refuses, saying only "Can't position this here..."
This PR adds additional tooltip info
Forest Frontiers 2024-05-06 23-06-15

### CODE REVIEW

**Comment by harvito** [comment]
Fixes #21974

**Comment by SpartanFrederic104** [comment]
This deserves a changelog entry, maybe something like this:
`- Fix: [#21974] No reason specified when attempting to place benches, lamps, or bins on path with no unconnected edges (original bug).`

File: data/language/en-GB.txt [hunk 1]
```diff
@@ -3699,6 +3699,7 @@ STR_6624    :Tile Inspector: Sort elements
 STR_6625    :Invalid colour
 STR_6626    :Animation is backwards
 STR_6627    :Track speed too high!
+STR_6628    :No unconnected edges available!
```
Comment (Gymnasiast) [inline data/language/en-GB.txt:3702]: Maybe this is a little bit clearer?

```suggestion
STR_6628    :Can only be placed on path edges!
```

Comment (harvito) [reply data/language/en-GB.txt]: done

**Comment by harvito** [comment]
@Gymnasiast I applied your suggestion, please rereview

File: src/openrct2/localisation/StringIds.h [hunk 2]
```diff
@@ -4047,6 +4047,8 @@ enum : uint16_t
 
     STR_SPEED_TOO_HIGH = 6627,
 
+    STR_CANNOT_PLACE_NO_UNCONNECTED_EDGES_AVAILABLE = 6628,
```
Comment (Gymnasiast) [inline src/openrct2/localisation/StringIds.h:4050]:
```suggestion
    STR_CAN_ONLY_BE_PLACED_ON_PATH_EDGES = 6628,
```

File: src/openrct2/actions/FootpathAdditionPlaceAction.cpp [hunk 3]
```diff
@@ -121,7 +121,9 @@ GameActions::Result FootpathAdditionPlaceAction::Query() const
     if (!(sceneryFlags & (PATH_ADDITION_FLAG_JUMPING_FOUNTAIN_WATER | PATH_ADDITION_FLAG_JUMPING_FOUNTAIN_SNOW))
         && (pathElement->GetEdges()) == 0x0F)
     {
-        return GameActions::Result(GameActions::Status::InvalidParameters, STR_CANT_POSITION_THIS_HERE, STR_NONE);
+        return GameActions::Result(
+            GameActions::Status::InvalidParameters, STR_CANT_POSITION_THIS_HERE,
+            STR_CANNOT_PLACE_NO_UNCONNECTED_EDGES_AVAILABLE);
```
Comment (Gymnasiast) [inline src/openrct2/actions/FootpathAdditionPlaceAction.cpp:126]:
```suggestion
            STR_CAN_ONLY_BE_PLACED_ON_PATH_EDGES);
```

**Review by Gymnasiast — APPROVED** [review]

**Pull request merged**